tests/python/contrib/test_hexagon/conftest.py

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+""" Hexagon testing fixtures used to deduce testing argument
+    values from testing parameters """
+
+import tvm
+from .infrastructure import ceildiv
+
+
+@tvm.testing.fixture
+def shape_nhwc(batch, in_channel, in_size):
+    return (batch, in_size, in_size, in_channel)
+
+
+@tvm.testing.fixture
+def shape_oihw(out_channel, in_channel, kernel):
+    return (out_channel, in_channel, kernel, kernel)
+
+
+@tvm.testing.fixture
+def shape_oihw8i32o4i(out_channel, in_channel, kernel):
```

Comment:
trying to parse this out: 8i 32o 4i
Meaning in factor is 8, out factor is 32 and second/inner in factor is 4. Am I reading this right?

Reply:
This is for filters with logical shape: OIHW, where O = out_channels, I = in_channels, and H and W are the kernel dimensions. `oihw8i32o4i` can be read as [O//32, I//32, H, W, 8i, 32o, 4i], where the out_channel dim is split into two axes (O//32, and 32o), and the in_channel dim is split into three axes [I//32, 8i, 4i].

Reply:
Aha, so in this case `in_factor = in_first_factor * in_second_factor`. Got it, thank you.